### PR TITLE
Remove mesa18 and libosmesa

### PR DESCRIPTION
**Pull request by alalazo**
refers to #44184

`mesa18` was introduced in #19528 as a way to maintain the old autotools build of mesa separate from the new meson build.

We could add a second build system to `mesa`, but since `mesa18` has been deprecated for a long time, we'll just remove it.

`libosmesa` was used to multiplex the `gl` provider between `mesa18` and `mesa`, and is thus unnecessary. Remove it to reduce complexity in the graphical stack.

### CODE REVIEW

**Comment by alalazo** [comment]
@kwryankrattiger I plan to simplify further the stack, since I think multiplexing using virtuals, when a package provides two implementation of the same virtual, is unnecessary - and just adds complexity. I'll ping you also in subsequent PRs. Let me know if I should add other reviewers too.

**Comment by kwryankrattiger** [comment]
Just quickly paged through this. I want to do a little more testing but this looks good so far.

**Comment by alalazo** [comment]
@kwryankrattiger Unfortunately this fails in packages using the virtuals, so I guess I need to fix them in this PR too. I'll try to keep the commits well separated to ease review.

**Comment by alalazo** [comment]
@kwryankrattiger If gitlab passes, I think this one is ready to go.

File: var/spack/repos/builtin/packages/vtk/package.py [hunk 1]
```diff
@@ -51,7 +51,6 @@ class Vtk(CMakePackage):
 
     # VTK7 defaults to OpenGL2 rendering backend
     variant("opengl2", default=True, description="Enable OpenGL2 backend")
-    variant("osmesa", default=False, description="Enable OSMesa support")
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/vtk/package.py:54]: So looking at this again, we want to be able to specify this variant for the case we build X and OSMesa versions in the same environment. If VTK is a root spec, that is no longer possible. Ref. ParaView and VisIt

Comment (alalazo) [reply var/spack/repos/builtin/packages/vtk/package.py:54]: Not sure I understand this use case. Modulo the fact that the `depends_on` directives on develop are currently wrong, I assume the intended behavior for `+osmesa` was to select `osmesa` as a `gl` provider. If my understanding is correct, then the variant is not necessary, and one can simply say:
```
vtk ^[virtuals=gl] osmesa
vtk ^[virtuals=gl] glx
```
if the environment is not unified, or set the `gl` provider in configuration if it is:
```yaml
packages:
  gl:
    require:
    - osmesa
```

Comment (kwryankrattiger) [reply var/spack/repos/builtin/packages/vtk/package.py:54]: Ah right, I see that now. I think you explained that to me before but my brain was still lagging a bit.

Okay. I think this looks good. Let's review the CI concrete environment, but I don't anticipate much issue.

**Comment by kwryankrattiger** [comment]
The `paraview/bfkhwvguli7a5bttlx62ozqxwhx7cfxc` is not  correct still with this patch. We are still getting a `paraview+osmesa ^glx` which I would like to fix with this patch if possible.

**Comment by alalazo** [comment]
Can we get this one merged, and fix `paraview` in a following PR? I'd like to have smaller scoped PRs, rather than changes all over the places. This one seems nicely scoped:
- Remove an unneeded virtual
- Remove `mesa18`

**Comment by alalazo** [comment]
To be clear, I'm already working on fixing the packages with the `osmesa` variant, but would like to keep this change as a standalone.

**Comment by kwryankrattiger** [comment]
> Can we get this one merged, and fix paraview in a following PR?

I only ask to fix it here because we do for visit and currently the binaries we produce in CI for the paraview+osmesa package are borked. But we can do it in a follow up too if the fix is more extensive the the visit fix.

> To be clear, I'm already working on fixing the packages with the osmesa variant, but would like to keep this change as a standalone.

I agree that fixing any other  OSMesa packages not in CI can be pushed to another PR.

**Review by kwryankrattiger — APPROVED** [review]
This looks like a good start, thanks!

**Pull request merged**